src/ledger.js

```diff
@@ -170,12 +169,26 @@ export class LedgerInteraction extends DirectKeystoreInteraction {
       const transport = await TransportWebUsb.create();
       return callback(transport);
     } catch (e) {
-      if (e.message && e.message === "navigator.usb is undefined") { // webusb failed, so try u2f
-        const transport = await TransportU2F.create();
-        return callback(transport);
-      } else {
-        throw new Error("Error with Ledger transport");
+      if (e.message) {
+        //WebUSB Errors:
+        // If you hit cancel on the browser webusb popup
+        if (e.message === 'No device selected.') {
+          e.message = `Select your device in the WebUSB dialog box. Make sure it's plugged in, unlocked, and has the Bitcoin app open.`;
+        }
+        if (e.message === 'undefined is not an object (evaluating \'navigator.usb.getDevices\')') {
+          e.message = `Safari is not a supported browser.`;
+        }
```

Comment:
Should we be using `e.message`?  Seems like `statusCode` or `statusText` (badly named) would be less fragile:

Example:
```js
{
  name: "TransportStatusError",
  message: "Ledger device: Condition of use not satisfied (denied by the user?) (0x6985)",
  stack: "Error↵    at new Je (https://waldenraines.github.i…k.js?__WB_REVISION__=2f69d2e5e4a3a6900d27:2:6804)",
  statusCode: 27013,
  statusText: "CONDITIONS_OF_USE_NOT_SATISFIED"
}
```

Reply:
Huh, yeah it seems weird to reassign the property of the error thrown. At the very least, either duplicating to make a new error object, or just a new constant `message` that is thrown at the end. 

I think it would be really good if we had some custom Error types. This would definitely be out of scope for this PR, but something that I encountered as potentially useful when I was developing in caravan too.

Reply:
The errors you're listing are not being sanitized in any way from the calling application (caravan test suite) - so those are the actual messages that come back from the device interactions.

Reply:
Agreed, statusCode generally is more reliable.  I think we had gotten in the habit of message checking b/c the didn't initially pass along error codes

Reply:
@nikkirostollan  -- The status codes and text will be visible once we're actually interfacing with the device. The error's we're attempting to catch here in calling the transport are because we cannot yet talk to the device at all.

E.g. error object looks like this, so message is what we have to work with:
![Screen Shot 2020-07-15 at 10 12 35 AM](https://user-images.githubusercontent.com/10471648/87574784-b94bf600-c683-11ea-8b38-0576b5d0e1ce.png)

Reply:
Ahh makes sense.

Reply:
For completeness: 
firefox error -> ![Screen Shot 2020-07-15 at 10 19 25 AM](https://user-images.githubusercontent.com/10471648/87575373-ac7bd200-c684-11ea-9f06-dbbb9325e385.png)
cancel on webusb -> 
![Screen Shot 2020-07-15 at 10 20 44 AM](https://user-images.githubusercontent.com/10471648/87575517-e6e56f00-c684-11ea-8eb1-3eddefd6d0b6.png)

Reply:
Sorry for the confusion, I was just saying that checking `message` instead of `statusCode` (or whatever) seemed fragile because they could change the message at any point but the `statusCode` seemed less likely to change.

I'm not seeing the same errors you are in webusb.  

Device locked:
```js
{
  name: "TransportStatusError",
  message: "Ledger device: Internal error, please report (0x6f04)",
  stack: "Error↵    at new Je (https://waldenraines.github.i….js?__WB_REVISION__=0583eaf57b022e5b05f7:2:10903)",
  statusCode: 28420,
  statusText: "UNKNOWN_ERROR"
}
```

Clicking cancel on web usb dialog:
```js
Error: Error with Ledger transport
    at 2.170be0ce.chunk.js?__WB_REVISION__=0583eaf57b022e5b05f7:2
    at c (2.170be0ce.chunk.js?__WB_REVISION__=0583eaf57b022e5b05f7:2)
    at Generator._invoke (2.170be0ce.chunk.js?__WB_REVISION__=0583eaf57b022e5b05f7:2)
    at Generator.throw (2.170be0ce.chunk.js?__WB_REVISION__=0583eaf57b022e5b05f7:2)
    at p (2.170be0ce.chunk.js?__WB_REVISION__=0583eaf57b022e5b05f7:2)
    at s (2.170be0ce.chunk.js?__WB_REVISION__=0583eaf57b022e5b05f7:2)
```

Bitcoin app not open:

```js
{
  name: "TransportStatusError",
  message: "Ledger device: INS_NOT_SUPPORTED (0x6d00)",
  stack: "Error↵    at new Je (https://waldenraines.github.i….js?__WB_REVISION__=0583eaf57b022e5b05f7:2:10903)",
  statusCode: 27904,
  statusText: "INS_NOT_SUPPORTED"
}
```

Rejected sign:

```js
{
  name: "TransportStatusError",
  message: "Ledger device: Condition of use not satisfied (denied by the user?) (0x6985)",
  stack: "Error↵    at new Je (https://waldenraines.github.i….js?__WB_REVISION__=0583eaf57b022e5b05f7:2:10903)",
  statusCode: 27013,
  statusText: "CONDITIONS_OF_USE_NOT_SATISFIED"
}
```

Etc.


How did you cause the error in the screenshot?

Reply:
you need to run unchained-wallets locally, and then console.log the error to see it, e.g.
```
async withTransport(callback) {
    try {
      const transport = await TransportWebUsb.create();
      return callback(transport);
    } catch (e) {
      console.log(e);
      throw new Error(e);
    }
  }
```

There are two forms of webusb failures:  ones where you're already paired to the device, and ones where you're not yet paired to a device.  Also note: webusb can actually look like three separate *devices* to the browser: 

1. Pair with unlocked but on dashboard
2. Pair with Bitcoin app open
3. Pair with Bitcoin Testnet app open

The errors I am showing are when you're not yet paired with anything or when you're paired with unlock+dashboard and trying to do things ( ). The errors you're seeing is when you're paired with a bitcoin/testnet app and you're seeing actual errors.

To remove pairings, click on "Not Secure" in Chrome and click the x's next to the device pairings.
![Screen Shot 2020-07-15 at 11 08 57 AM](https://user-images.githubusercontent.com/10471648/87580389-4b57fc80-c68c-11ea-8237-262aeee5012a.png)

As for `Error with Ledger transport` - that comes from https://github.com/unchained-capital/unchained-wallets/blob/v0.1.3/src/ledger.js#L177

The point is that the pre-paired errors (and only those) are what we're handling here. The post-paired errors are handled by the application calling it and account for almost all of the errors you are posting.

Reply:
Basically, anything that's happening in `withTransport` right since that's what's setting up the pairing?

Reply:
Yes - the other errors are firing from successfully using the `transport` but unsuccessfully talking to the particular `app`.